### PR TITLE
Refactor: move NavModel to @grafana/ui

### PR DESCRIPTION
**Pull request by ryantxu**
In #16586, I want to give app plugins more control of the page layout (the header, title, tabs, etc)

This moves NavModel to @grafana/ui, it also:
* consolidates 2 versions of NavModel into one (previously we had an interface and a class)
* adds types for `NavModelBreadcrumb`

Most of the PR is updates to import paths

### CODE REVIEW

File: packages/grafana-ui/src/types/navModel.ts [hunk 1]
```diff
@@ -1,22 +1,29 @@
 export interface NavModelItem {
   text: string;
-  url: string;
+  url?: string;
   subTitle?: string;
+  subType?: any;
```
Comment (torkelo) [inline packages/grafana-ui/src/types/navModel.ts]: can't see this being used anywhere.

Comment (torkelo) [reply packages/grafana-ui/src/types/navModel.ts]: probably a legacy from old plugin or data source pages. Think the subTitle is all that is actually used, and seems sufficient. Can remove this from here (and PageHeader.tsx)

**Review by torkelo — APPROVED** [review]
Looks good. Just a comment about a subType that i think can be removed.

File: packages/grafana-ui/src/types/navModel.ts (hunk 1 above)
Comment (ryantxu) [reply packages/grafana-ui/src/types/navModel.ts]: Removed:
```
 {main.subType && (
            <div className="page-header__stamps">
              <i className={main.subType.icon} />
              {main.subType.text}
            </div>
          )}
```

waiting for this to pass CI, then will merge

**Pull request merged**